halo2_proofs/src/plonk/circuit.rs

```diff
@@ -59,9 +59,51 @@ impl<C: ColumnType> PartialOrd for Column<C> {
     }
 }
 
+/// Phase of advice column
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub enum Phase {
```

Comment:
I think I also prefer the marker type approach.

Reply:
I'm also dubious about restricting this to three phases. Why three?

Reply:
I thought about expose phase as `struct Phase(u8)` like rotation, but I was afraid it might look too flexible, so I choose to use enum, and set it to have 3 phases from my naive imagination.

But since we are going to expose the phases with exhaustive types, we might be able to define internal phase as `struct Phase(u8)` now? Then it should make it much easier to add more phases.